tests/tep/test_upsert_validator.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+
+from bigchaindb.models import Transaction
+
+pytestmark = [pytest.mark.tendermint, pytest.mark.bdb]
+
+
+def test_upsert_validator_valid_election(validator_election, priv_validator_path,
+                                         network_validators58, new_validator):
+
+    # Create an election proposal to add a new validator
+    tx_election = validator_election.propose(new_validator, priv_validator_path)
+
+    voters = {}
+    for output in tx_election.outputs:
+        [output_public_key] = output.public_keys
+        voters[output_public_key] = output.amount
+
+    valid_asset = {
+        'type': 'election',
+        'name': 'upsert-validator',
+        'version': '1.0',
+        'args': new_validator
```

Comment:
Makes sense to describe the validator anew or make a copy of it here, otherwise the test won't notice any undesired changes made to the `new_validator` object inside `propose`.

Reply:
I thought you were suggesting to create a `deepcopy` of `new_validator`? Its a little defensive but I made the change.

Reply:
Sorry, that's not what I meant - copy it in the test, like:

```
'version': '1.0',
'args': {
    'public_key': ..
..
```

In the code I would not be defensive without a good reason - it can significantly affect the performance.